hbase-procedure/src/main/java/org/apache/hadoop/hbase/procedure2/util/DelayedUtil.java

```diff
@@ -79,7 +79,7 @@ public String toString() {
    */
   public static <E extends Delayed> E takeWithoutInterrupt(final DelayQueue<E> queue) {
     try {
-      return queue.take();
+      return queue.poll(15, TimeUnit.SECONDS);
```

Comment:
Just change the method to poll and let the upper layer pass in the timeout parameters?

Reply:
Although it sounds good, I believe so far we have only 2 callers and both can pass 15s. That should be fine? Let me change this.